### PR TITLE
Add test for a union using an alias with scope 

### PR DESCRIPTION
**Pull request by richiware**
## Description

Add tests for the regression type for issue \#21823 in eprosima/fast-dds-gen#411.

Depends on:

- eprosima/dds-types-test#39

@Mergifyio backport 3.1.x 3.0.x

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- *N/A* Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- *N/A* Changes are API compatible. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [x] Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/dds/xtypes/update_header_and_create_cases.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def __init__(self):
 
     def extract_structures(self, idl_text):
         # Regular expressions for module and struct extraction
-        module_regexp = r'module\s+(\w+)\s*{((?:.|[\r\n])*?)\};\s*\};'
+        module_regexp = r'module\s+(\w+)\s*{((?:.|[\r\n])*?)\};'
```
Comment (MiguelCompany) [inline test/dds/xtypes/update_header_and_create_cases.py:24]: Why this change?

File: include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobject.hpp [hunk 2]
```diff
@@ -399,7 +399,11 @@ class TypeObjectHashId
                     }
 
                     selected_member_ = 0x00000001;
-                    member_destructor_ = [&]() {m_hash.~EquivalenceHash();};
+                    member_destructor_ = [&]()
+                    {
+                        using namespace eprosima::fastdds::dds::xtypes;
+                        m_hash.~EquivalenceHash();\
```
Comment (MiguelCompany) [inline include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobject.hpp:405]: Code needs to be regenerated

File: thirdparty/dds-types-test
Comment (MiguelCompany) [inline thirdparty/dds-types-test:1]: Submodule needs to be updated.

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: test/dds-types-test/unionsPubSubTypes.cxx [hunk 3]
```diff
@@ -30,6 +30,51 @@
...
+
+
+
+
+
+
+
+
+
+
```
[the remark this answers is not in the document]
Comment (richiware) [reply test/dds-types-test/unionsPubSubTypes.cxx]: Done in 50124f8

File: include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobject.hpp (hunk 2 above)
Comment (richiware) [reply include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobject.hpp]: Done in 50124f8ed6

**Comment by MiguelCompany** [comment]
@Mergifyio backport 3.1.x 3.0.x

**Pull request merged**